app/serializers/current_user_serializer.rb

```diff
@@ -153,9 +156,26 @@ def custom_fields
   end
 
   def muted_category_ids
-    @muted_category_ids ||= CategoryUser.where(user_id: object.id,
-                                               notification_level: TopicUser.notification_levels[:muted])
-      .pluck(:category_id)
+    CategoryUser.lookup(object, :muted).pluck(:category_id)
+  end
+
+  def top_category_ids
+    categories = CategoryUser.where(user_id: object.id)
+      .where.not(notification_level: CategoryUser.notification_levels[:muted])
+      .select(:category_id, :notification_level)
+
+    categories.sort_by do |c|
+      case c[:notification_level]
+      when CategoryUser.notification_levels[:watching]
+        1
+      when CategoryUser.notification_levels[:tracking]
+        2
+      when CategoryUser.notification_levels[:watching_first_post]
+        3
+      end
+    end
+
+    categories.pluck(:category_id).slice(0, MAX_TOP_CATEGORIES_COUNT)
```

Comment:
this is a second query... 

instead what you do is order by a case statement in SQL no need to pull the data down